network-libp2p/src/network.rs

```diff
@@ -303,16 +304,34 @@ impl Network {
     fn new_transport(
         keypair: &Keypair,
         memory_transport: bool,
+        tls: &Option<TlsConfig>,
     ) -> std::io::Result<Boxed<(PeerId, StreamMuxerBox)>> {
         if memory_transport {
             // Memory transport primary for testing
             // TODO: Use websocket over the memory transport
 
             #[cfg(feature = "tokio-websocket")]
-            let transport = websocket::WsConfig::new(dns::TokioDnsConfig::system(
+            let mut transport = websocket::WsConfig::new(dns::TokioDnsConfig::system(
                 tcp::tokio::Transport::new(tcp::Config::default().nodelay(true)),
-            )?)
-            .or_transport(MemoryTransport::default());
+            )?);
+
+            // Configure TLS if the configuration has the corresponding entry
+            #[cfg(feature = "tokio-websocket")]
+            if let Some(tls) = tls {
+                let priv_key = websocket::tls::PrivateKey::new(tls.private_key.clone());
+                let certificates: Vec<_> = tls
+                    .certificates
+                    .clone()
+                    .into_iter()
+                    .map(websocket::tls::Certificate::new)
+                    .collect();
+                transport
+                    .set_tls_config(websocket::tls::Config::new(priv_key, certificates).unwrap());
```

Comment:
Is it save to use `unwrap()` here? Or is panicking on purpose if the config fails?

Reply:
I saw that the bytes have already been verified when they were read from the file :+1: So it's safe to unwrap here, but if it panics, then that is what we want anyway.